src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,487 @@
+  - ProcessLoginCallbackForWsFed(result, additionalLocalClaims, localSignInProps);
+  - ProcessLoginCallbackForSaml2p(result, additionalLocalClaims, localSignInProps);
+  + ProcessLoginCallback(result, additionalLocalClaims, localSignInProps);
+  ```
+
+* `ConsentResponse` does not contain a definition for `ScopesConsented`:
+
+  ```diff lang="csharp" title="*.cs"
+  grantedConsent = new ConsentResponse
+  {
+      RememberConsent = model.RememberConsent,
+  -    ScopesConsented = scopes.ToArray()
+  +    ScopesValuesConsented = scopes.ToArray()
+  };
+  ```
+
+### Step 3: Update Database Schema
+
+If you are using the `IdentityServer4.EntityFramework` package to store configuration and operational data in a [database](../../data),
+you'll need to create two database migrations that update the database schema.
+Note that you may want to change the database migration paths in the examples below to reflect your project structure.
+
+:::caution
+Make sure to verify the database migrations as part of your IdentityServer4 upgrade, to ensure all configuration and operational data remains in place.
+:::
+
+For the operational data, you can create and apply an Entity Framework Core migration that targets the `PersistedGrantDbContext` database context.
+
+<Steps>
+
+1. Create the migration:
+
+   ```bash title="Terminal"
+   dotnet ef migrations add Grants_v4 -c PersistedGrantDbContext -o Migrations/PersistedGrantDb
+   ```
+
+2. Apply the migration to your database:
+
+   ```bash title="Terminal"
+   dotnet ef database update -c PersistedGrantDbContext
+   ```
+
+</Steps>
+
+For your configuration data, the conversation from the old schema to the new will need to be performed by applying a custom SQL script.
+We'll start with creating a migration that targets the `ConfigurationDbContext` database context:
+
+<Steps>
+
+1. Create the migration:
+
+   ```bash title="Terminal"
+   dotnet ef migrations add Config_v4 -c ConfigurationDbContext -o Migrations/ConfigurationDb
+   ```
+
+   You will see a message _"An operation was scaffolded that may result in the loss of data. Please review the migration for accuracy."_ in the output.
+   To avoid data loss, the migration will need to be updated.
+
+2. To ensure no data is lost, make sure to include the [`ConfigurationDb_v4_delta.sql`](https://github.com/DuendeArchive/UpgradeSample-IdentityServer4-v3/blob/main/IdentityServerMigrationSample/ConfigurationDb_v4_delta.sql)
+   script in your project.
+
+   You can add the script as an embedded resource by updating the `.csproj` file:
+
+   ```xml title=".csproj"
+   <ItemGroup>
+       <EmbeddedResource Include="ConfigurationDb_v4_delta.sql" />
+   </ItemGroup>
+   ```
+
+   :::note[Update the SQL script for your database type]
+   The `ConfigurationDb_v4_delta.sql` file assumes you are using SQL Server. If a different database server type is used for your IdentityServer host, you'll need to update the SQL script to use the correct syntax.
```

Comment:
Maybe we should provide variations for PostgreSQL, MySQL, and Oracle. Just a side note.